src/osd/osd_types.h

```diff
@@ -194,7 +195,7 @@ struct pg_shard_t {
   void dump(ceph::Formatter *f) const {
     f->dump_unsigned("osd", osd);
     if (shard != shard_id_t::NO_SHARD) {
-      f->dump_unsigned("shard", shard);
+      f->dump_unsigned("shard", static_cast<int>(shard));
```

Comment:
static cast to int here, and to uint (which I think is the correct one) on line 612?